streams/src/test/java/org/apache/kafka/streams/processor/internals/metrics/ProcessorNodeMetricsTest.java

```diff
@@ -0,0 +1,393 @@
+        final String descriptionOfMaxLatency = "The maximum latency of creations of processor nodes";
+        shouldGetThroughputAndLatencySensorWithParentOrEmptySensor(
+            metricNamePrefix,
+            descriptionOfRate,
+            descriptionOfCount,
+            descriptionOfAvgLatency,
+            descriptionOfMaxLatency,
+            () -> ProcessorNodeMetrics.createSensor(THREAD_ID, TASK_ID, PROCESSOR_NODE_ID, streamsMetrics)
+        );
+    }
+
+    @Test
+    public void shouldGetDestroySensor() {
+        final String metricNamePrefix = "destroy";
+        final String descriptionOfCount = "The total number of processor nodes destroyed";
+        final String descriptionOfRate = "The average number of processor nodes destroyed per second";
+        final String descriptionOfAvgLatency = "The average latency of destructions of processor nodes";
+        final String descriptionOfMaxLatency = "The maximum latency of destructions of processor nodes";
+        shouldGetThroughputAndLatencySensorWithParentOrEmptySensor(
+            metricNamePrefix,
+            descriptionOfRate,
+            descriptionOfCount,
+            descriptionOfAvgLatency,
+            descriptionOfMaxLatency,
+            () -> ProcessorNodeMetrics.destroySensor(THREAD_ID, TASK_ID, PROCESSOR_NODE_ID, streamsMetrics)
+        );
+    }
+
+    @Test
+    public void shouldGetForwardSensor() {
+        final String metricNamePrefix = "forward";
+        final String descriptionOfCount = "The total number of calls to forward";
+        final String descriptionOfRate = "The average number of calls to forward per second";
+        setUpThroughputParentSensor(
+            metricNamePrefix,
+            descriptionOfRate,
+            descriptionOfCount
+        );
+        setUpThroughputSensor(
+            metricNamePrefix,
+            descriptionOfRate,
+            descriptionOfCount,
+            RecordingLevel.DEBUG,
+            expectedParentSensor
+        );
+
+        verifySensor(() -> ProcessorNodeMetrics.forwardSensor(THREAD_ID, TASK_ID, PROCESSOR_NODE_ID, streamsMetrics));
+    }
+
+    @Test
+    public void shouldGetLateRecordDropSensor() {
+        final String metricNamePrefix = "late-record-drop";
+        final String descriptionOfCount = "The total number of dropped late records";
+        final String descriptionOfRate = "The average number of dropped late records per second";
+        if (builtInMetricsVersion == Version.FROM_0100_TO_24) {
+            setUpThroughputSensor(
+                metricNamePrefix,
+                descriptionOfRate,
+                descriptionOfCount,
+                RecordingLevel.INFO
+            );
+        } else {
+            expect(streamsMetrics.nodeLevelSensor(
+                THREAD_ID,
+                TASK_ID,
+                PROCESSOR_NODE_ID,
+                metricNamePrefix,
+                RecordingLevel.INFO
+            )).andReturn(expectedSensor);
```

Comment:
What is `expectedSensor` used here since it is just a mock? I feel we can just state that we expect it to be called but not specify returning anything?

Reply:
Sounds good.